HHS/Head-Start-TTADP
# [TTAHUB-3345] Add filters to QA dashboard, frontend

## Description of change
Adds filters to the QA dashboard. The size of this change is mainly due to refactoring - other instances of our ```useFilters``` hook had to be updated. The upside of this is that more logic is now internal to the hook and we have less repeated code on the various dashboards.

This change also incorporated some updates to allow a series of pages to share filters while only surfacing some to the UI depending on the page's configuration.

One page is yet to be coded but once it is, it should be a simple drop-in/configuration to get it added.

Finally, I added an "SSDI" data fetcher that will strip unneeded key/values from the filter object before sending a request to the API. Given the uncertain state of the backend, we may need to refactor that but it will likely still be useful given the shared filters

## How to test
- No data is being fetched from the backend. This is just a test to see if the filters function as expected pre-data-fetching.
- Visit the QA dashboard.
- Add some filters. Visit one of the subpages, change some filters.
- Note that adding a filter persists between pages (if both pages have it) as does deleting.
- Note also that if a page does not have a filter present in another, it disappears on the page that doesn't have it but reappears when you return.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3345

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
